proposals/0043-groupshared-arguments.md

```diff
@@ -29,12 +30,47 @@ parameter declarations. The keyword when applied to a parameter declaration of
 type `T`, alters the qualified type of the parameter to a `groupshared T &`
 (a reference to `groupshared` memory of type `T`).
 
+```c++
+void fn(groupshared uint4 A) {}
+```
+
 No implicit or explicit conversion can change the memory space of an object. To
 perform such a conversion, a user must declare a new object in the destination
 memory space and initialize it appropriately. For overload resolution, the
 parameter type must be an exact match in order for overload resolution to
 succeed since no conversions will be valid.
 
+Allowed:
+```c++
+void fn(groupshared uint4 A) {
+  float4 LocalA = (float4) A;
+  doesSomething(LocalA);
+  A = (uint4) LocalA;
+}
+```
+
+Not Allowed:
+```c++
+void fn(groupshared uint4 A) {}
+void fn2() {
+  float4 B = 1.0.xxxx;
+  fn(B); // Error:
+  fn((uint4)B); // Error:
+}
+```
+
+There is a mostly working proof of concept here (does not compile library
+shaders):
+[Proof of Concept](https://github.com/spall/DirectXShaderCompiler/tree/groupsharedParam).
```

Comment:
You'll have to remember never to delete that branch!

I wonder if https://github.com/microsoft/DirectXShaderCompiler/tree/262b6ff5b61aabe882528ad8d57a865262317406 would be a longer-lasting link?